src/bin/common/parse_command.ml

```diff
@@ -0,0 +1,1157 @@
+    Arg.(value & flag & info ["no-minimal-bj"] ~docs ~doc) in
+
+  let no_sat_learning =
+    let doc =
+      "Disable learning/caching of unit facts in the Default SAT. These \
+       facts are used to improve bcp." in
+    Arg.(value & flag & info ["no-sat-learning"] ~docs ~doc) in
+
+  let no_tableaux_cdcl_in_instantiation =
+    let doc = "When satML is used, this disables the use of a tableaux-like\
+               method for instantiations with the CDCL solver." in
+    Arg.(value & flag &
+         info ["no-tableaux-cdcl-in-instantiation"] ~docs ~doc) in
+
+  let no_tableaux_cdcl_in_theories =
+    let doc = "When satML is used, this disables the use of a tableaux-like\
+               method for theories with the CDCL solver." in
+    Arg.(value & flag & info ["no-tableaux-cdcl-in-theories"] ~docs ~doc) in
+
+  let sat_plugin =
+    let doc =
+      "Use the given SAT-solver instead of the default DFS-based SAT solver." in
+    Arg.(value & opt string (get_sat_plugin ()) &
+         info ["sat-plugin"] ~docs ~doc) in
+
+  let sat_solver =
+    let doc = Format.sprintf
+        "Choose the SAT solver to use. Default value is CDCL (i.e. satML \
+         solver). Possible options are %s."
+        (Arg.doc_alts ["CDCL"; "satML"; "CDCL-Tableaux";
+                       "satML-Tableaux"; "Tableaux-CDCL"])
+    in
+    let docv = "SAT" in
+    Arg.(value & opt string "CDCL-Tableaux" &
+         info ["sat-solver"] ~docv ~docs ~doc) in
+
+  Term.(ret (const mk_sat_opt $
+             get_bottom_classes $ disable_flat_formulas_simplification $
+             enable_restarts $ no_arith_matching $
+             no_backjumping $ no_backward $ no_decisions $ no_decisions_on $
+             no_minimal_bj $ no_sat_learning $
+             no_tableaux_cdcl_in_instantiation $
+             no_tableaux_cdcl_in_theories $ sat_plugin $ sat_solver
+            ))
+
+let parse_term_opt =
+
+  let docs = s_term in
+
+  let disable_ites =
+    let doc = "Disable handling of ite(s) on terms in the backend." in
+    Arg.(value & flag & info ["disable-ites"] ~docs ~doc) in
+
+  let inline_lets =
+    let doc =
+      "Enable substitution of variables bounds by Let. The default \
+       behavior is to only substitute variables that are bound to a \
+       constant, or that appear at most once." in
+    Arg.(value & flag & info ["inline-lets"] ~docs ~doc) in
+
+  let rewriting =
+    let doc = "Use rewriting instead of axiomatic approach." in
+    Arg.(value & flag & info ["rwt"; "rewriting"] ~docs ~doc) in
+
+  let term_like_pp =
+    let doc = "Output semantic values as terms." in
+    Arg.(value & flag & info ["term-like-pp"] ~docs ~doc) in
+
+  Term.(ret (const mk_term_opt $
+             disable_ites $ inline_lets $ rewriting $ term_like_pp
+            ))
+
+let parse_theory_opt =
+
+  let docs = s_theory in
+
+  let disable_adts =
+    let doc = "Disable Algebraic Datatypes theory." in
```

Comment:
Does this also prevents datatypes from being typed ? In any case, the behavior wrt to typing should be documented here.

Reply:
I don't know